### PR TITLE
Use `runZonedGuarded` when using an error handler. 

### PR DESCRIPTION
**Pull request by lrhn**
Needed for NNBD SDK compatibility.

@kevmoo Not sure who owns this package now.

### CODE REVIEW

**Comment by kevmoo** [comment]
@lrhn – please check travis.

We'll have to bump the min SDK it looks like? Update .travis.yml, etc

**Comment by lrhn** [comment]
There is no way Travis can actually work until we publish a dev-SDK with the breaking change included.
At that point the SDK requirement should be upped to that dev-SDK version.

This branch is intended to be used by the SDK itself, referenced by hash, because building the SDK depends on this package.

File: pubspec.yaml [hunk 1]
```diff
@@ -7,14 +7,14 @@ name: stack_trace
...
-version: 1.9.3
+version: 1.9.4
 
 description: A package for manipulating stack traces and printing them readably.
 author: 'Dart Team <misc@dartlang.org>'
 homepage: https://github.com/dart-lang/stack_trace
 
 environment:
-  sdk: '>=1.23.0 <3.0.0'
+  sdk: '>=2.8.0-dev <3.0.0'
```
Comment (kevmoo) [inline pubspec.yaml:17]: It's totally valid to have a lower-bound of both the SDK here and in .travis.yaml of the SDK that ships the feature you're using!

**Comment by kevmoo** [comment]
> There is no way Travis can actually work until we publish a dev-SDK with the breaking change included.
> At that point the SDK requirement should be upped to that dev-SDK version.
> 
> This branch is intended to be used by the SDK itself, referenced by hash, because building the SDK depends on this package.

This feature hasn't shipped yet?

**Comment by natebosch** [comment]
My understanding was that the old API would continue to work in code that is not opted in to NNBD - were we unable to do that?

**Comment by jakemac53** [comment]
Can we add the new feature in a dev SDK prior to making the breaking change? That is the typical way to roll out a breaking change like this.

**Comment by lrhn** [comment]
The issue was that the `onError` was removed in the NNBD SDK, but the SDK itself seems to depend on this package, so I couldn't build the NNBD SDK.
I'll add the `onError` back in to the NNBD SDK until we have migrated packages off it.

**Comment by natebosch** [comment]
This PR doesn't quite go far enough, there would still be static errors. Superseded by https://github.com/dart-lang/stack_trace/pull/93

**Pull request closed without merging**